### PR TITLE
Add get_download_series() to retrieve download data from BQ

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/15011

---

This is similar to what we have for usage stats. We rely on the table until we
have a productionized view in BigQuery.

### CODE REVIEW

File: src/olympia/stats/utils.py [hunk 1]
```diff
@@ -6,35 +6,52 @@
...
     'apps': 'dau_by_app_version, dau_by_fenix_build',
     'countries': 'dau_by_country',
     'locales': 'dau_by_locale',
     'os': 'dau_by_app_os',
     'versions': 'dau_by_addon_version',
 }
 
+# This is the mapping between the AMO download stats `sources` and the BigQuery
+# columns.
+AMO_TO_BQ_INSTALLS_COLUMN_MAPPING = {'sources': 'downloads_per_source'}
```
Comment (willdurand) [inline src/olympia/stats/utils.py:21]: We'll have more entries later (when we'll implement new pages)

File: src/olympia/stats/tests/test_utils.py [hunk 2]
```diff
@@ -435,3 +477,85 @@ def test_returned_values(self, bigquery_client_mock):
...
+AND submission_date BETWEEN @submission_date_start AND @submission_date_end
+ORDER BY submission_date DESC
+LIMIT 365"""
+
+        get_download_series(
+            addon=self.addon, start_date=start_date, end_date=end_date
+        )
+
+        client.query.assert_called_once_with(
+            expected_query, job_config=mock.ANY
```
Comment (bobsilverberg) [inline src/olympia/stats/tests/test_utils.py:526]: Shouldn't we also verify the `query_parameters` part of `job_config`? That seems just as important as verifying that the `query` was correct.

Comment (willdurand) [reply src/olympia/stats/tests/test_utils.py:526]: Yep, good idea. I didn't have a good way to assert `job_config` back then but now I have a helper so I'll update this test case and the similar one for `get_updates_series`

File: src/olympia/stats/tests/test_utils.py [hunk 3]
```diff
@@ -150,6 +159,39 @@ def test_filter_by_dau_by_app_version_and_no_fenix_data(self):
         }
 
 
+class TestRowsToSeriesForInstallStats(BigQueryTestMixin, TestCase):
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/olympia/stats/tests/test_utils.py:162]: I'll update to use downloads but yeah both terms are interchangeable.

File: src/olympia/stats/tests/test_utils.py [hunk 4]
```diff
@@ -435,3 +477,85 @@ def test_returned_values(self, bigquery_client_mock):
...
+
+    @mock.patch('olympia.stats.utils.statsd.timer')
+    @mock.patch('google.cloud.bigquery.Client')
+    def test_create_query(self, bigquery_client_mock, timer_mock):
+        client = self.create_mock_client()
+        bigquery_client_mock.from_service_account_json.return_value = client
+        start_date = date(2020, 5, 27)
+        end_date = date(2020, 5, 28)
+        expected_query = f"""
+SELECT submission_date, total_downloads
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/olympia/stats/tests/test_utils.py:514]: No.

[the remark this answers is not in the document]
Comment (willdurand) [reply src/olympia/stats/tests/test_utils.py:514]: > Is there a way to test the actual query itself

No

> and not only test that the expected query was passed? How do we know there isn't a bug in the query itself?

- Copy/paste in Redash-
- Manual testing locally
- QA in -dev

File: src/olympia/stats/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (willdurand) [reply src/olympia/stats/utils.py:21]: yep

**Pull request merged**